ecosystem/typescript/sdk_v2/src/core/account_address.ts

```diff
@@ -0,0 +1,342 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+import { bytesToHex, hexToBytes } from "@noble/hashes/utils";
+import { Serializer, Deserializer } from "../bcs";
+import { HexInput } from "../types";
+import { ParsingError, ParsingResult } from "./common";
+
+/**
+ * This enum is used to explain why an address was invalid.
+ */
+export enum AddressInvalidReason {
+  INCORRECT_NUMBER_OF_BYTES = "incorrect_number of bytes",
+  INVALID_HEX_CHARS = "invalid_hex_chars",
+  TOO_SHORT = "too_short",
+  TOO_LONG = "too_long",
+  LEADING_ZERO_X_REQUIRED = "leading_zero_x_required",
+  LONG_FORM_REQUIRED_UNLESS_SPECIAL = "long_form_required_unless_special",
+}
+
+/**
+ * NOTE: Only use this class for account addresses. For other hex data, e.g. transaction
+ * hashes, use the Hex class.
+ *
+ * AccountAddress is used for working with account addresses. Account addresses, when
+ * represented as a string, generally look like these examples:
+ * - 0x1
+ * - 0xaa86fe99004361f747f91342ca13c426ca0cccb0c1217677180c9493bad6ef0c
+ *
+ * Proper formatting and parsing of account addresses is defined by AIP-40.
+ * To learn more about the standard, read the AIP here:
+ * https://github.com/aptos-foundation/AIPs/blob/main/aips/aip-40.md.
+ *
+ * The comments in this class make frequent reference to the LONG and SHORT formats,
+ * as well as "special" addresses. To learn what these refer to see AIP-40.
+ */
+export class AccountAddress {
+  /*
+   * This is the internal representation of an account address.
+   */
+  readonly data: Uint8Array;
+
+  /*
+   * The number of bytes that make up an account address.
+   */
+  static readonly LENGTH: number = 32;
+
+  /*
+   * The length of an address string in LONG form without a leading 0x.
+   */
+  static readonly LONG_STRING_LENGTH: number = 64;
+
+  static ADDRESS_ONE: AccountAddress = AccountAddress.fromString({ str: "0x1" });
+
+  static ADDRESS_TWO: AccountAddress = AccountAddress.fromString({ str: "0x2" });
+
+  static ADDRESS_THREE: AccountAddress = AccountAddress.fromString({ str: "0x3" });
+
+  static ADDRESS_FOUR: AccountAddress = AccountAddress.fromString({ str: "0x4" });
```

Comment:
I forget can statics be used in Typescript like `AccountAddress.ONE`, that may be easier then.

Reply:
Good call let me do that.